### ISSUE TITLE
Api doesn't responds

### ISSUE THREAD

**Issue by amadeuszblanik** [natewiley/cpv2api#14]
Seems like all requests return an error for the last few days.

**Comment by rainner** [comment]
Same here. All the endpoints are returning 503 Service Unavailable.

**Comment by alexzaworski** [comment]
```
SyntaxError: Unexpected token < in JSON at position 0
    at JSON.parse (<anonymous>)
    at Request._callback (/Users/alexzaworski/Desktop/cpv2api/app.js:118:72)
    at Request.self.callback (/Users/alexzaworski/Desktop/cpv2api/node_modules/request/request.js:185:22)
    at emitTwo (events.js:126:13)
    at Request.emit (events.js:214:7)
    at Request.<anonymous> (/Users/alexzaworski/Desktop/cpv2api/node_modules/request/request.js:1161:10)
    at emitOne (events.js:116:13)
    at Request.emit (events.js:211:7)
    at IncomingMessage.<anonymous> (/Users/alexzaworski/Desktop/cpv2api/node_modules/request/request.js:1083:12)
    at Object.onceWrapper (events.js:313:30)
Worker 18986 died with code: 1, and signal: null
Starting a new worker
undefined:1
<!DOCTYPE html>
^
```

here's the error, should be fixable but I'm not sure if Nate is still maintaining this which means there's no way for us to deploy changes :(

**Comment by natewiley** [comment]
My apologies, this is something that I'm aware of and have been wanting to fix, I just have a lot of things going right now and haven't found the time to get to it. 

@alexzaworski thank you for looking into the cause, PRs are welcome on this and I can send you an invite on the repo.

**Comment by alexzaworski** [comment]
@natewiley! Great to hear from you

If I have time this week I'll absolutely look into this, if nothing else I can look at it over the weekend. Also down to help/cheerlead if someone else wants to take it.

**Comment by rainner** [comment]
Hey @alexzaworski  @natewiley 

The problem seems to be related to cloudflare. The responses from the codepen endpoints are coming back with a captcha page, that's why the JSON.parse is failing. One solution I tried that works is to make a custom wrapper for the request module being used to send the requests that looks at the "set-cookie" response header from cloudflare, find the "__cfduid" cookie value and send the request back with that cookie in the headers. 

I'll try to do a PR later.

**Comment by alexzaworski** [comment]
@rainner it sounds like you have more experience in this area than I do so it's quite likely you've stumbled into this already but [here's a lib](https://github.com/codemanki/cloudscraper) that may help

**Comment by alexzaworski** [comment]
There's an additional question here of "should we actually be doing this," though. May want to hit up CodePen support and see if they're okay with the api taking steps to bypass a captcha.

edit: I'll hit them up and post a response here

**Comment by rainner** [comment]
Yeah might be best to check with them on this. Might be faster to send Chris a tweet: https://twitter.com/chriscoyier

**Comment by alexzaworski** [comment]
I just emailed them— they're usually pretty quick to respond. Seems better to handle privately anyways imo

**Comment by alexzaworski** [comment]
Heard back, they were totally understanding and cool with it as long as we're not doing anything too crazy (just use your judgment). 

@rainner if you want to take this go for it, otherwise (or if you want help) feel free to ping me

**Comment by alexzaworski** [comment]
closed by #16 <3
